devlab/tests/test_verify_dst_functionality.py

```diff
@@ -0,0 +1,318 @@
+import tests.config as config
+import tests.functional_test as functional_test
+import time
+import pprint
+from fabric.api import run, sudo, local, settings
+from neutronclient.common.exceptions import NeutronClientException
+from novaclient.exceptions import Forbidden
+from novaclient.exceptions import ClientException as NovaClientBaseException
+from cinderclient.exceptions import ClientException as CinderClientBaseException
+from keystoneclient.exceptions import \
+    ClientException as KeystoneClientBaseException
+from glanceclient.common.exceptions import \
+    BaseException as GlanceClientBaseException
+
+
+TIMEOUT = 300
+TEST_TENANT_NAME = 'tenant4'
+TEST_VM_NAME = 'VMtoVerifyDstCloudFunc'
+
+CINDER_VOLUME_CHK_CMDS = ['mkdir /tmp/test',
+                          '/usr/sbin/mkfs.ext2 /dev/vdb',
+                          'mount /dev/vdb /tmp/test',
+                          'touch /tmp/test/newfile']
+
+CINDER_VOLUME_PARAMS = {'display_name': 'tn4_volume5',
+                        'size': 1,
+                        'server_to_attach': None,
+                        'device': '/dev/vdb'}
+SSH_CHECK_CMD = 'pwd'
+
+
+class VerifyDstCloudFunctionality(functional_test.FunctionalTest):
+
+    def setUp(self):
+        # some initial variables needed for teardown
+        self.keypair_list_to_delete = []
+
+        self.dst_cloud.switch_user(user=self.dst_cloud.username,
+                                   password=self.dst_cloud.password,
+                                   tenant=TEST_TENANT_NAME)
+        self.dst_tenant_id = str(
+            self.dst_cloud.get_tenant_id(TEST_TENANT_NAME))
+
+        # declare vars in case if condition below will not be satisfied
```

Comment:
If you assume than condition below can be not satisfied, you have to handle this case